internal/css_parser/css_parser.go

```diff
@@ -828,11 +828,21 @@ abortRuleParser:
 	case "keyframes", "-webkit-keyframes", "-moz-keyframes", "-ms-keyframes", "-o-keyframes":
 		p.eat(css_lexer.TWhitespace)
 		var name string
+		var isStringName bool
 
 		if p.peek(css_lexer.TIdent) {
-			name = p.decoded()
+			name = strings.ToLower(p.decoded())
+			if name == "none" {
+				p.log.AddID(logger.MsgID_CSS_CSSSyntaxError, logger.Warning, &p.tracker, atRange, "Expected identifier but found `none` which is invalid in this context")
+				p.prevError = p.current().Range.Loc
+			}
+			p.advance()
+			isStringName = false
+		} else if p.peek(css_lexer.TString) {
+			name = strings.ToLower(p.decoded())
+			isStringName = !p.options.MinifySyntax || cssWideAndReservedKeywords[name] || name == "none"
 			p.advance()
-		} else if !p.expect(css_lexer.TIdent) && !p.eat(css_lexer.TString) && !p.peek(css_lexer.TOpenBrace) {
+		} else if !(p.expect(css_lexer.TIdent) && p.expect(css_lexer.TString)) && !p.peek(css_lexer.TOpenBrace) {
 			// Consider string names a syntax error even though they are allowed by
 			// the specification and they work in Firefox because they do not work in
 			// Chrome or Safari.
```

Comment:
Not sure if you saw this comment or not but lack of support for this was deliberate. This is currently a Firefox-only feature, which I don't think makes sense to support natively in esbuild at this time.

I agree that the bug that changes the contents of `@keyframes` rules with unsupported syntax should still be fixed. These should be passed through unmodified. I will fix that part.

Reply:
I currently working on blink, and the webkit one is already merged

ref: https://github.com/WebKit/WebKit/pull/4234
      https://chromium-review.googlesource.com/c/chromium/src/+/3865188

Reply:
This is not a Firefox-only syntax, the prefixed `@-webkit-keyframes / @-moz-keyframes` always support `<string>`, and there are [UseCounter](https://codereview.chromium.org/651123003/) in Chrome that say: `The usage counter for strings in prefixed case is too high`.